core/embed/rust/src/crypto/mod.rs

```diff
@@ -16,9 +18,9 @@ pub enum Error {
 impl From<Error> for crate::error::Error {
     fn from(e: Error) -> Self {
         match e {
-            Error::SignatureVerificationFailed => value_error!("Signature verification failed"),
-            Error::InvalidEncoding => value_error!("Invalid key or signature encoding"),
-            Error::InvalidParams => value_error!("Invalid cryptographic parameters"),
+            Error::SignatureVerificationFailed => value_error!(c"Signature verification failed"),
+            Error::InvalidEncoding => value_error!(c"Invalid key or signature encoding"),
+            Error::InvalidParams => value_error!(c"Invalid cryptographic parameters"),
```

Comment:
I wonder if it would be possible to avoid specifying the c-literal at every occurrence of `value_error!()`. Perhaps it could be incorporated into the macro itself, but I am not sure if it's possible at zero cost.

Reply:
i'm pretty sure the macro system is not strong enough for it: `c"blabla"` is a single token, you can't take out (or add) the `c` at start.

Previously `value_error!` used the (3rd party) `cstr!` macro, which works by explicitly concatting a `\0` at end of the provided string, and then calling an unchecked conversion. It didn't seem worth keeping it, if the cost is "write a value_error!, see compile failure, remember to add `c` at start")

(I also looked into why we need a `CStr` here in the first place. the reason: if the incoming string is null-terminated, we don't have to copy it onto the micropython heap, we only need to allocate one block for the wrapping `str` object. This will pretty much always succeed if the GC itself is still working.)

Reply:
yes, I was afraid it wouldn't be possible. I don't think it's a problem to solve.